app/components/x-list.js

```diff
@@ -154,7 +158,7 @@ export default Component.extend({
    * @method setupContextMenu
    */
   setupContextMenu() {
-    let menu = this.resizableColumns.getColumnVisibility().reduce((arr, { id, name, visible }) => {
+    this.menuContent = this.resizableColumns.getColumnVisibility().reduce((arr, { id, name, visible }) => {
```

Comment:
Shouldn't we just have `getColumnVisibility` figure out when to show/hide columns, rather than passing a column name to every list-cell?

Reply:
@rwwagner90 I think the problem we'd run into is that we don't use a data-driven approach to render the cell contents in most cases, except for in the `record` view.  https://github.com/emberjs/ember-inspector/pull/805/files#diff-1f81a1f049b22867a84780b0a7f9885eR21 is pretty much how we do it -- we manually wrap all the columns we want to render with an `x-list-cell`.

No matter what as the responsibility of showing which columns to show, we need some way of feeding that information down to our individual templates (at least with how they're implemented now)
We can't fall back to just doing things positionally either.  I'm afraid one way or another we need a way to say "this td belongs to this column" in the templates.

Reply:
Sounds logical. I will defer the decision to @teddyzeenny as he likely knows a lot more about it than me 😄